xlml/apis/task.py

```diff
@@ -69,14 +69,30 @@ def run(self) -> DAGNode:
     with TaskGroup(
         group_id=self.task_test_config.benchmark_id, prefix_group_id=True
     ) as group:
-      provision, queued_resource, ssh_keys = self.provision()
+      provision, queued_resource, ssh_keys, _ = self.provision()
```

Comment:
Can you merge `run_with_gcs_name_generation` with `run` if the only difference is making the GCS path available? AFAICT `generate_gcs_file_location` doesn't do anything destructive, and downstream tasks can decide whether or not they need to write metrics to GCS

Reply:
If so, the env variable will always been passing to the run_model and provision will assume `gcs_location` is consumed:
```
run_model = self.run_model(
          queued_resource, ssh_keys, {f"{metric_config.MetricPath.GCS_OUTPUT}": gcs_location}
      )
```
Do we want to keep `run` and `run_with_gcs_name_generation` separate or we want to pass argument like `specify_gcs_location:bool` to `run()`?

Reply:
Yeah, I think that's fine. Consumers that don't need the extra env won't use it